src/main/scala/nodes/learning/KernelRidgeRegression.scala

```diff
@@ -0,0 +1,343 @@
+package nodes.learning
+
+import workflow.LabelEstimator
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+import edu.berkeley.cs.amplab.mlmatrix.RowPartitionedMatrix
+import edu.berkeley.cs.amplab.mlmatrix.util.{Utils => MLMatrixUtils}
+
+import scala.util.Random
+
+class KernelRidgeRegression(
+  kernelBlockGenerator: KernelGenerator,
```

Comment:
Yeah, I think I like this. My mental model:
`KernelEstimator: Estimator[DenseVector[Double], Matrix]`  (basically does nothing but pass along a reference to the input RDD), .fit returns a KernelGenerator
`KernelGenerator: Transformer[DenseVector[Double], Matrix]` (does the work of forming the Kernel matrix - note that this can be applied to training or test data)
Then the output of the KernelGenerator is passed to BlockLeastSquaresEstimator.
So the whole pipeline is as simple as 
`Preprocessing andThen (KernelEstimator, train) andThen (BlockLeastSquaresEstimator, train, labels)`

We can wrap this all in a `Pipeline` object (or a function that takes an RDD and returns a `Pipeline` or something) so that the whole thing can be chained in the same way we chain a vanilla linear solver to the end of a pipeline.

Does this make sense to both of you @shivaram @Vaishaal